### ISSUE TITLE
Evaluate options for project stewardship, non-profit

### ISSUE THREAD

**Issue by eloquence** [rethinkdb/rethinkdb#6137]
A straightforward possibility is to apply for membership in the Software Freedom Conservancy, which offers important services such as the ability to accept donations:

https://sfconservancy.org/members/services/

Another path would be to join Apache:
https://incubator.apache.org/

(This would require relicensing, if I'm interpreting https://incubator.apache.org/ correctly, so should probably be taken care of ASAP if that's the path you want to take.)

**Comment by dustinkingen** [comment]
Another possibility is joining The Eclipse Foundation.

https://eclipse.org/projects/handbook/#starting

**Comment by mglukhovsky** [comment]
This is something we're exploring right now. Advice and introductions are welcome: mglukhovsky@gmail.com

**Comment by nirgn975** [comment]
I think [Linux Foundation](https://www.linuxfoundation.org/projects) should be an option too.

**Comment by lenovouser** [comment]
Linux Foundation sounds nice, not sure about Apache. They already have a lot of databases ([Cassandra](https://cassandra.apache.org/), [CouchDB](https://couchdb.apache.org/)) - which makes me unsure about how much time will be actually spent on continuing developing RethinkDB. Also loads of their projects who also had a decent amount of users started getting unmaintained for a few years or still are (e.g. [Shindig](https://shindig.apache.org/) or [Archiva](https://archiva.apache.org/)). Maybe even @stripe could support RethinkDB as they [contribute a lot](https://stripe.com/open-source) to Open Source already.

**Comment by v3ss0n** [comment]
I am willing to contribute 5% of every project i have done to Rethinkdb. Please keep it going. I am a startup founder too . I know there will be hard times. But What we are doing with RethinkDB are big!

**Comment by dvergeylen** [comment]
I am exactly in the same situation of @v3ss0n.

Maybe crowdfunded bug resolutions / feature requests could provide cash for short/mid term?
I personnally would be ready to fund some features I was waiting for or at least contribute to help your developers. 

Thank you for what you have done over these 7 years! You are inspiring.

**Comment by bcantrill** [comment]
Speaking personally, but as a member of the [CNCF Technical Oversight Committee](https://cncf.io/about/technical-oversight-committee), I would love to see RethinkDB in the CNCF -- and I think it would be a great fit.  I do think that the AGPL may pose an obstacle, so it would be helpful to know if relicensing (to Apache 2) would be considered on the table -- but even if that were not an option, I would love to try to find a way for RethinkDB to be a CNCF project!

**Comment by babakness** [comment]
A Patreon option? Other Open Source projects are using it with success and get both community and corporate support. Example:

https://www.patreon.com/evanyou

Acknowledgement of top donors is a nice bonus. Here is another example of this

https://www.apache.org/foundation/sponsorship.html

More directly: POSTGRESQL

https://www.postgresql.org/about/sponsors/

Perhaps Stripe can offer something where they don't charge the 3% and give it all to the project in addition to an official sponsorship. 

I'd like the see a license change to BDS, MIT, or Apache as well.

**Comment by tracker1** [comment]
As an addendum to what a lot of people have mentioned regarding AGPL for the server... it would be nice to see something like MIT or ISC for the client libraries.  Right now, the npm module for rethinkdb doesn't list any license.

Beyond that, it seems to be that a push towards an Apache license may be for the best as to the server regarding being able to support it in a foundational org.  I think that @bcantrill is correct in that it would be a good fit within the CNCF.

**Comment by brianm** [comment]
A fundamental difference between Linux Foundation (LF) foundations (`a la the Node.js Foundation and CNCF) and the Apache Software Foundation is the unit of participation at the foundation level. The LF is geared for pay-to-play corporate running of the foundation, which can be, and usually is, totally separate from the technical operation of projects therein. The ASF is business friendly but always explicitly composed of the individuals who contribute, no the corporations. Those individuals may be paid to work on the project, but control of the project and foundation is to the individual person.

If you want to be able to have the _foundation_ employee people to work on the project, go with the LF (possibly via CNCF, but I am not involved in the CNCF so I don't know if it is set up to allow for that). 

To the comment that the ASF has databases already, so it is unclear "how much time will be actually spent on continuing developing RethinkDB" -- the ASF doesn't devote people to working on projects, it provides a fairly well-proven governance model geared towards long-term viability of projects. It doesn't work well for all projects, and is no guarantee of the long-term viability, but has a good track record overall. That is about it :-)

LF foundations are much more flexible in how things work. The LF foundation package provides a framework for how things can work, legal and administrative assistance, and then basically lets the foundation go do its thing. If you want your own foundation and are prepared to drum up corporate sponsor/members, the LF is WAY easier than forming a new foundation from scratch and is about the best way to go. You will need to develop your own governance model, but if you want your own governance model, you'll need that anyway!

I am not familiar with the model set up by the CNCF (an LF foundation), so cannot compare it.

**Comment by v3ss0n** [comment]
I raised the AGPL concern a year ago, A few google employees asked before that too that AGPL should  be dropped to increase adoption because at Google they can't use RethinkDB , due to AGPL.

https://groups.google.com/d/msg/rethinkdb/g5UEck3sqMs/nDkcltNIB4kJ
https://groups.google.com/d/msg/rethinkdb/g5UEck3sqMs/2ZCUW92TwPQJ

**Comment by brianm** [comment]
Regardless of its intent, the awkward and unclear phrasing in the AGPL keeps a large number of companies from touching it :-(

**Comment by eloquence** [comment]
@brianm This is super helpful - I hope the rethinkdb folks read it. I think the choice is between these two models indeed:

- an independent org (through LF, Conservancy or otherwise) that hires and employs developers but also works with corporate partners + individual contributors
- a project within an established structure like Apache that works with contributors of all stripes but does not itself hire developers.

Which of these two models is the right one likely depends on whether there is a champion within the current contributor community who is able to drive the first one. Sustainably paying FTEs through a non-profit is hard as those who've pushed this model in the infrastructure space (e.g., MariaDB) can attest. There's arguably higher risk / higher reward here.

If the org goes for the second approach, I would recommend engaging a contractor to shepherd the process, at least for Apache which is fairly complex. I can suggest a few folks who I think could take this on, and others might too. But the licensing piece would have to be cleared up before anything else.

**Comment by v3ss0n** [comment]
Most orgs turns away just by looking at AGPLV3 licence , regardless of what it does. I think that is why RethinkDB was being avoided in many cases. 
Also startups would afraid if they are planning for buyout strategy. Big guys will just say know if depend on AGPL.

**Comment by ShaneCurcuru** [comment]
A few comments from a long-time ASF Member:

- The openZipkin community is considering a move to a foundation right now as well, and there are a few gems of useful information for RethinkDB in there:

https://github.com/openzipkin/openzipkin.github.io/issues/51

michaelsembwever and myself have some good data there about Apache; caniszczyk provides some good data about CNCF at the LF.

- The ASF only uses the Apache license v2.0.  So you'd need to decide ahead of time if you were able and willing to relicense to Apache before making a proposal to the Apache Incubator.

- If you have some coherent community organization, but want the minimal overhead and restrictions, then consider SPI or the Software Freedom Conservancy.  They don't offer the community and infrastructure support that CNCF or ASF would offer, but they provide a legal place to hang your hat and do fundraising - and own your trademarks - with very broad ability to decide how you run your own project governance.

The real question is: are there enough people active in the community to spend at least a little time **working together** to document and organize the move to a new home?  That's along with showing that there's still a vibrant developer community interested in continuing to work on the code.

Various LF Collaborative Projects (like CNCF) and the ASF have plenty of mentors who do have experience with how our foundations work, so please do reach out to us if you have questions.

- Shane

**Comment by teslan** [comment]
Yet another open-source scam?

This whole incident is an excellent illustration of how much scam or scam-like activity is going on within open source. OK, I just read what I wrote and it hurts to read what I just wrote, so then let me try to explain why I would inflict so much pain onto myself and likely others.

If RethinkDB ever was an open source project then now there would be no questions about need for new licensing nor its continuity as an open source project. All of that would have already been in place. There are too many "open source" projects [like this] that pretend to be open source but are first and foremost geared towards making a profit for those that have invested in it. There is nothing wrong with making a profit and since there has never been anything like a free lunch, profits must be made because someone must end up paying for it all.

However, the accompanying commercial business plans must be based on added service that are on top of a thriving open source community. Those add-on services must not be replacements for at least basic admin tools nor for poor documentation, which too often is lacking because of the add-on services. If those add-on services can not deliver added value, in form of better quality and/or compressed development timelines because more qualified experts would be doing the work then there is no add-on value to speak of.

I have been personally struggling (at 55+++) to learn and keep up, navigating through all of these web development mine fields and a web development toolbox whose contents are just about totally replaced every year or two. Nevertheless, even I feel that I could come up with multiple ways in which I could make a profit, only if these tools would settle down a bit. In fact, are we all not here because we feel that we can do the same thing? So, how is it that those who were running RethinkDB as a business, with all that inside info and those $12 MILLION raised, that they were unable to direct a tiny portion of all of those resources into ensuring a sustained self-existence?

Take care and shut the lights out behind you.

**Comment by babakness** [comment]
@teslan RethinkDB is awesome. I believe you need to research what a scam is. This is more like reverse scam, we gained not lost. RethinkDB created millions of dollars worth of value and people used this product for free to make money out of their start-ups / projects / etc. I'm look forward to seeing this thrive as a self-sustained an open source project.

**Comment by eloquence** [comment]
In any event, it's off-topic here. @teslan - if you want to discuss that further, please move it elsewhere. Thanks.

**Comment by v3ss0n** [comment]
12 mil$ is very low for this team of 20 top notch developers with great teamwork developing a technologically superior NOSQL database for 7 years . RethinkDB is invested by YC investors who know verywell what they are doing , Nobody can scam that.  So please drop it  , or delete it @teslan .

**Comment by v3ss0n** [comment]
> However, the accompanying commercial business plans must be based on added service that are on top of a thriving open source community. Those add-on services must not be replacements for at least basic admin tools nor for poor documentation, which too often is lacking because of the add-on services. If those add-on services can not deliver added value, in form of better quality and/or compressed development timelines because more qualified experts would be doing the work then there is no add-on value to speak of.

RethinkDB never goes a Fremium model with limiting functionality , they give every thing since public version. They openly communicates with the community on every issues , features and decision.

**Comment by sjmueller** [comment]
Perhaps there's value in @mglukhovsky and @coffeemug commenting on the possibilities of changing the license from AGPL to something more liberal? It'd be really helpful to understand the potential downsides, since we've been hearing quite a bit about the upsides during the past few days.

I think from the community's perspective (especially those still using the software), there is a real sense of danger that rethinkdb might not be viable 1-2 years down the road, especially if the right support isn't garnered while there's a ton of attention on the product.

As a founder, there's something to be said about momentum. Sometimes it's this intangible thing that gets lost during times of volatility, yet it is a fiercely important component of success that is hard to maintain and 10x harder to restart after it is lost -- not something to be taken for granted. If there are orgs that can shepherd rethinkdb into a sustainable place again without losing too much momentum, that'd be huge for both the legacy of the rethinkdb team, as well as the confidence of all those that use and love this beautiful piece of software. If the license is identified as a blocker, let us know the specific reasons why a change isn't the immediate consideration here.

**Comment by sirinath** [comment]
I think you should look at moving to the ASF. This will ensure continuity and governance for the community since the organisation backing the project is closing down.

**Comment by gbaz** [comment]
SPI in my understanding does provide similar services to SFC, to  a number of orgs: http://www.spi-inc.org/projects/

We've been affiliated them for haskell.org for some years. They only provide the 501(c)3 umbrella and some basic services (and not necessarily very rapidly, to be honest, but that's due to their own resource constraints), but have been pretty decent at doing what they're supposed to, and its been much appreciated. I can't speak to the tradeoffs between them and the SFC very well, but they're worth mentioning, I suppose.

**Comment by sirinath** [comment]
Has any decision being reached about this?

**Comment by kofalt** [comment]
My 2 cents, since there's a lot of community input here: I would greatly prefer RDB remain AGPL.

It would be counter-productive to elaborate: I really, **really** don't want to derail this thread into a debate about licensing merits. Suffice it to say that I'm read into the subject & familiar with the ramifications :smile: 

Critically, note that the drivers are [already Apache licensed](https://github.com/rethinkdb/rethinkdb/blob/next/drivers/COPYRIGHT), and Slava has a [relevant blurb](https://groups.google.com/d/msg/rethinkdb/g5UEck3sqMs/ZDBsiZ0tUdAJ) on this as well.

**Comment by srh** [comment]
@teslan It's poor form to copy/paste your rant on multiple websites.

**Comment by v3ss0n** [comment]
We have listed the problem AGPL
- Many Big Guys won't use it , Just take a google search of which companies doesn't allow AGPL, almost all big tech companies, less contribution in terms of both funding and code
- Less employees using RethinkDB , which cause same problem above.
- Startups aiming for Exit Strategy Have to avoid it , no choice!

AGPL dosen't help anyone. In practice those more liberal licence got much much funding from big guys.

**Comment by v3ss0n** [comment]
@kofalt  https://groups.google.com/d/msg/rethinkdb/g5UEck3sqMs/spBSWG4xVIcJ as you can see there Dan wants to change since before company shut down.

**Comment by tracker1** [comment]
@kofalt Thanls for the client license reference, raised a PR to note it in the package.json for the js/npm version, so it would show up on npm.

I have mixed feelings as to AGPL though, while I appreciate the protections of agpl, I'm not sure that the defense against a customized SaaSified version of RethinkDB really has nearly as much merit if the company behind the project is shutting down anyway.  GPL (without the A) has been working fine for MySQL/MariaDB adoption.  So maybe a straight reduction to GPL 2 or 3 might be okay...

Either way, I have no plans to customize the codebase, or start a SaaS around the product... just being understanding of some of the concerns of others.

**Comment by sirinath** [comment]
I think best line of strategy is identify what organisations would be looking to continue the development open source and what their licensing preference is and governance model is. Prematurely deciding these will be the death of the project.

**Comment by v3ss0n** [comment]
@sirinath  thanks a lot for your insights!

**Comment by ha1331** [comment]
> Yet another open-source scam?
> 
> This whole incident is an excellent illustration of how much scam or scam-like activity is going on within open source. OK, I just read what I wrote and it hurts to read what I just wrote, so then let me try to explain why I would inflict so much pain onto myself and likely others.
>
> 
> If RethinkDB ever was an open source project then now there would be no questions about need for new licensing nor its continuity as an open source project. All of that would have already been in place. There are too many "open source" projects [like this] that pretend to be open source but are first and foremost geared towards making a profit for those that have invested in it. There is nothing wrong with making a profit and since there has never been anything like a free lunch, profits must be made because someone must end up paying for it all.

Hurts me too, but I suspect for different reasons. I find it hard to follow the reasoning, at all. I wonder if you actually mean something else, Could you explain how is rethinkdb not open source or less now than it used to be? How have you or anyone else been scammed? Did you think it was under different license than AGPL? Do you feel betrayed because it was? Or do you think AGPL is not a valid open source license? 

Rethink has been AGPL for as long as I know, Still is. So what is the issue you would like to fix or prefer to have been fixed before the company shut down? Not going to touch the "there would be no questions about need for new licensing" part, but I'm curious to know how could have they ensured that there will be no questions about it's continuity as an open source project? Do tell if you have figured that out. I suspect it's too late for these products, but I bet there are ton of people building products that could use your advice. So, what would you have had in place, have you been mr RethinkDB and had the reigns?

> However, the accompanying commercial business plans must be based on added service that are on top of a thriving open source community. Those add-on services must not be replacements for at least basic admin tools nor for poor documentation, which too often is lacking because of the add-on services. If those add-on services can not deliver added value, in form of better quality and/or compressed development timelines because more qualified experts would be doing the work then there is no add-on value to speak of
> 
> I have been personally struggling (at 55+++) to learn and keep up, navigating through all of these web development mine fields and a web development toolbox whose contents are just about totally replaced every year or two. Nevertheless, even I feel that I could come up with multiple ways in which I could make a profit, only if these tools would settle down a bit. In fact, are we all not here because we feel that we can do the same thing? So, how is it that those who were running RethinkDB as a business, with all that inside info and those $12 MILLION raised, that they were unable to direct a tiny portion of all of those resources into ensuring a sustained self-existence?

What was the point of all the rest? You're 55 and have hard time keeping up with bunch of stuff, but you know how other people should have done bunch of stuff or at least how they shouldn't have done it? How is this relevant, at all, to any of the discussion happening on the topic, which is, by the way: "Evaluate options for project stewardship, non-profit"? 

You make it sound like 12 MILLION is indicative of something, like its some huge amount of money and surely they should have had enough to use on stuff like ensuring the sustained self-existence. How would have that worked, they raise this money to build a business, people invested in it so they might some day make profit, but people behind RethinkDB the company should have siphoned some of that invested money to... what? Some foundation never existed, just to make sure that if it goes bad, bunch of that investors money is there sitting and waiting for the rainy day? Surely you can't mean anything like that?

I hope I don't come out as being terribly salty. Tho I do take offence in bunch of <redacted> when I personally feel like I got to use these products for free, enjoyed using them and hoped people behind said product "make it". Then when it didn't go like planned, these people had to shut down a company and kind of failed at something they used 7 years to build. After this, while we talk <redacted>, same bunch of people work to ensure these products have a future as a community driven effort. Even that's not enough, surely all this is just a scam, let's just call captain hindsight and talk bunch of nonsense about the nature of true open source... for reals mate?
> 
> Take care and shut the lights out behind you.

**Comment by v3ss0n** [comment]
They are working on it

**Comment by teslan** [comment]
Notwithstanding the fact that I could have used a better word nor the fact that none of this was likely done intentionally, it mind as well have been a scam because the final results are the same. Too many people started using it because of its "open source" label, without knowing what that really meant. I would expect so called open source projects, especially the funded commercial ones, to leverage and  incorporate their open source community into the way that they do business.  

Had any of that been done, today there would already be a community. There would be no need to be going back to square one and be talking about different types of licenses. The whole thing would have been less of an all-or-nothing proposition because there would already be at least some members whose food on the table comes from different companies, instead of everyone being dependent on a single source. 

The whole thing unraveled most likely because of difficulties in selling software licenses and/or services.  Instead of building software to sell, more and more companies are building software for themselves to use. RethinkDB as a service, would have made a fantastic open source alternative to Firebase. @ha1331, I would bet that you already knew all of that ;)

**Comment by ha1331** [comment]
> Notwithstanding the fact that I could have used a better word nor the fact that none of this was likely done intentionally, it mind as well have been a scam because the final results are the same. Too many people started using it because of its "open source" label, without knowing what that really meant. I would expect so called open source projects, especially the funded commercial ones, to leverage and incorporate their open source community into the way that they do business.
> 
You could have used a better word? Which one? You talk about open source, with the intensity of thousand suns, but to you it seems to not mean what you can do with the code base. It actually has very little to do with the code. To you it seems to be this idea of everything being there, people doing stuff for free, infinite funding, bunch of that funding is being pushed to make sure the project lives on. Not only  even that, the point you are making is, it's a scam because after a month from the announcement of the company shutting down, people are not pushing for the next release and the army of community welfare workers are not promising you the moon and the skies. You make absolutely no sense at all. 

I wonder if you even used RethinkDB or is this just another one of them things you have hard time keeping up with? I mean it's pretty bold to be saying there is no community around RethinkDB. U

> Had any of that been done, today there would already be a community. There would be no need to be going back to square one and be talking about different types of licenses. The whole thing would have been less of an all-or-nothing proposition because there would already be at least some members whose food on the table comes from different companies, instead of everyone being dependent on a single source.
> 
> The whole thing unraveled most likely because of difficulties in selling software licenses and/or services. Instead of building software to sell, more and more companies are building software for themselves to use. RethinkDB as a service, would have made a fantastic open source alternative to Firebase. @ha1331, I would bet that you already knew all of that ;)

You sir, are a meanie. People here are trying their best to sort out what license to switch to, what needs to happen before that, what foundation to go under, how to organize everything that was done by the people of the defunct company, so pretty much everything. What is your input in the matter? Well, scam, not proper open source, if you would have walked the Teslan road, people woudn't have to endure any of this, seeing how you would have done it properly. That is whole new level of ... something starts with an A and rhymes with a black hole. Out of respect to the community formed around these products, I'm just going to leave it at this. Live long and prosper brother @teslan .

**Comment by v3ss0n** [comment]
@teslan  Wait wft ? What is that? Did you even use RethinkDB before? Are you mistaken rethinkdb with some other freemium databases that scam the s*** out of you? Are you even Sane?
RethinkDB have over 100 contributors and had a full team of very dedicated and smart developers. Most of them even going to continue work on this.

**Comment by streaky** [comment]
> If RethinkDB ever was an open source project then now there would be no questions about need for new licensing nor its continuity as an open source project

You're confusing several different issues.

AGPL confuses people (including lawyers) - doesn't mean it an actual problem. The intent is fairly clear but people get lost and hung-up in the technicalities. It's hard to imagine that the license is the reason for Rethink's failure as a business. Either way it doesn't make the now defunct business was a scam.

What is true whoever is given it has failed and there's obviously support from users at least for a move to a more holistic open source approach to the IP that's there a more permissive license might help in the long run (and there's always going to be people who would say it might have helped 6 months ago).

Not for nothing but as what appears to be a non-contributer on a rant session I'd strongly question your motives.

**Comment by sirinath** [comment]
Also being part of a foundation (like ASF) will mean there is more commitment and contribution in the future and would be an assurance for corporate users.

**Comment by alexgorbatchev** [comment]
Curious if there has been any progress on this subject?

**Comment by marshall007** [comment]
@alexgorbatchev we have our next meeting this afternoon to discuss and get status updates from everyone in the leadership team. As always, we'll keep you guys posted in #open-rethinkdb on Slack, so stay tuned!

**Comment by kofalt** [comment]
Anyone mind transcribing updates from the Slack chan to this thread on occasion? Would be great :smile:

**Comment by babakness** [comment]
@kofalt 

There is a Google Doc now on this 

https://docs.google.com/document/d/1Fy-drBGDTyibwEBBAX6G5otqtWiKOD2sBXmL2uVTFtE/edit

**Comment by lbguilherme** [comment]
We need to move forward on this. No solution is far worse than any of the proposed solutions. RethinkDB as of now is stuck and losing momentum, it may soon be irreversible.

**Comment by ha1331** [comment]
@lbguilherme forward moving is being done as far as I know. Bunch of foundations have been contacted and people are evaluating them.

**Comment by v3ss0n** [comment]
@lbguilherme  @AtnNn  is working on the build server , its going great. we need testers for it in Windows and ARM platform.

**Comment by Morgul** [comment]
@marshall007 Is there any was we can get an update (either in this issue, or preferably a blog post on the site) just giving an overview of the current status, next steps, and ways the community can assist?

I love RethinkDB (it's my favorite database, period), and I am staunchly refusing to switch away from it, but I'd be lying if I said there wasn't a feeling of uncertainty when I started a new project last month and decided to stick with it. I fear that this is something where the process needs to be both transparent, and any updates shouted from the roof tops. Trying to wade through Slack history is a very unfortunate way for people to get updates.

If nothing else, could this (or whatever tracking issue you want to use) get added to the Readme, for easy discovery?

**Comment by v3ss0n** [comment]
@Morgul  please join https://rethinkdb.slack.com/messages/open-rethinkdb for latest updates and progress. RethinkDB development is Resuming and we are nearing for next release. 

Here are excrept from latest doc 
https://docs.google.com/document/d/1cTqKt1_EBanGoVmYyahdLyDD8dhCa0SdD0CbjbP67f8/edit

## New CI infrastructure is up and running (source: atnnn/rethinkdb-nix)

- Still missing packages for Yakkety, Centos, OS X, and Windows.
- Not currently running tests against OS X or Windows.

## Release v2.3.6: planned for next week (by 2017/02/03)
- #5878, #5426, #5300 - web UI broken in Microsoft Edge and Chrome 49
- #6084 - potential DoS vulnerability on ReQL client port
- #6090 - driver incompatibility with Ruby 1.9
- #6093 - segfault when indexes in use by open changefeeds are recreated
- #6148, #6194 - limit sample size to array size limit
- #6193 - don't persist stale env_t* in ordered unions
- #6198, #6219 - misc makefile/build fixes
## Release v2.4: planned for next week (by 2017/02/03)
- Everything initially staged by the official team with the addition of:
- #6191 - deterministic r.now fix
- #6215 - OpenSSL compatibility
- #6226 - Windows build error

**Comment by bryfry** [comment]
rethinkdb slack signup page: http://slack.rethinkdb.com/

**Comment by Morgul** [comment]
@v3ss0n Thanks for that update. I still think it would be really nice if some of this information could be added to the Readme, though. (If you want a PR I'm happy to do so.)

**Comment by v3ss0n** [comment]
yes please submit PR!

**Comment by coffeemug** [comment]
I decided not to participate in the open-rethinkdb project for personal reasons, but I'd like to add my opinion because there is a lot of misinformation going on (and IMO some suboptimal decisions are being made).

- There is a difference between licensing and IP.
- RethinkDB code is licensed under AGPL, drivers are Apache.
- One of the creditors owns the IP and IMO they're unlikely to sell it unless the project raises fairly substantial sums of money. If they do, great, but I wouldn't hold my breath. So I would proceed under that assumption.

In my opinion expending any resources whatsoever on buying the IP is a complete red herring. Plenty of projects operate without owning the IP (e.g. the Linux kernel). Admittedly, the kernel is licensed under GPL rather than AGPL, but the differences are mostly theoretical and are unlikely to play out in any practical way. The *actual* challenge of making open-rethinkdb succeed is in getting enough developers to contribute to it, simplifying/documenting the codebase to make it friendlier for new developers, and continuing shipping improvements to make the project useful for more and more people. Worrying about licensing is what PG would call a sitcom idea[1] -- it feels like doing useful work, but in actuality it makes no difference whatsoever.

If the project raises enough money to buy the IP from the creditors, I wouldn't spend a dime of that money on IP -- I'd give it to developers to encourage *actual* work on the project. IMO the discussion around keeping the name is also a red herring -- just call the project open-rethinkdb, put up a website, and focus on increasing the velocity of developer contributions. If someone threatens to sue (they won't, but in the unlikely event that they do), just rename it to something else and keep going.

Human psychology is structured in such a way that people naturally gravitate towards protecting themselves from failure. It's much easier to say to yourself "we failed because we couldn't buy back the IP from an uncooperative third party" than to say "we failed because we couldn't get people to contribute" or "we failed because we couldn't generate enough momentum behind the project". If I were doing this I would focus 100% of my energy on contribution velocity, and ignore the IP issue completely. It's a hundred times easier to solve any potential IP problems for a successful project than to solve the momentum problem even if the IP issues are all buttoned down.

[1] http://paulgraham.com/startupideas.html

**Comment by v3ss0n** [comment]
Thank you very much on this!Now we are quite clear what to do next.
>  The actual challenge of making open-rethinkdb succeed is in getting enough developers to contribute to it, simplifying/documenting the codebase to make it friendlier for new developers, and continuing shipping improvements to make the project useful for more and more people. Worrying about licensing is what PG would call a sitcom idea[1] -- it feels like doing useful work, but in actuality it makes no difference whatsoever.

We won't be able to move to foundation like Apache, so time to raise funding for developers ourselves i guess. Lets see.

**Comment by bcantrill** [comment]
To dismiss worrying about licensing -- especially when it's as toxic as the AGPL -- as a "sitcom idea" strikes me as tone deaf:  the ASF and LF/CNCF have already made clear that the AGPL is non-negotiable issue for them, as have many users here.  Moreover, the advice to "ignore the IP issue completely" is simply terrible life advice: just an artist should not "ignore contracts", software engineers should not ignore IP issues -- lest they wake up to discover their work hijacked by someone paying more attention than they.  This doesn't mean that software engineers should fixate on IP issues (or engage in endless and pointless debates over the merits of different open source licenses), but merely understand the fine print of the projects to which they contribute.  And in this case, the fine print is very, very important:  as it stands, the community doesn't know who owns RethinkDB or what their intent is -- just that the AGPL makes things like ownership and intent crucially important.

So if the IP issue isn't solved -- if it continues to be not only licensed under an infamously ill-defined and dangerously toxic license, but also owned by a shadowy creditor (or worse, resold to a potentially bad actor) -- there will not be "contribution velocity" because any developer would be a fool to contribute their time and energy to a project hanging under such a Damoclean sword.

**Comment by coffeemug** [comment]
Respectfully, you comment does much more to fuel the sort of FUD that would prevent developers from contributing than AGPL+IP situation combined. The argument strikes me as entirely pedantic -- plenty of companies that claimed to ban AGPL run vanilla Mongo because it's a useful project, and the creditor has no legal course (nor incentive) to hurt the open-source project. The Sword of Damocles only exists in the minds of a small but vocal minority.

Lack of knowhow of the codebase and questions around usecases required to garner a critical mass of users are much bigger, much more immediate, and much more important problems. The licensing situation strikes me as classic casuistry. I'm no longer involved with the project, but my advice to the leadership team would be to ignore it completely.

**Comment by sirinath** [comment]
Can the creditor be asked to donate the license?

**Comment by ha1331** [comment]
@sirinath well seeing how @coffeemug said: "One of the creditors owns the IP and IMO they're unlikely to sell it unless the project raises fairly substantial sums of money. If they do, great, but I wouldn't hold my breath. So I would proceed under that assumption." I assume the creditor can be asked bunch of things, but him/her actually donating would be fairly slim.

**Comment by sirinath** [comment]
If the creditor cannot make money why should he keep it. Better goodwill by donating.

**Comment by ha1331** [comment]
@sirinath I don't think that's the way the world works. If @coffeemug pretty much said they would only give it up for "substantial amount of money" you thinking they would just give it up for free because reasons... makes absolutely no sense in this context.

**Comment by coffeemug** [comment]
> If the creditor cannot make money why should he keep it. Better goodwill by donating.

It doesn't cost anything to hold the IP, and there is a minuscule but non-zero chance that some day the IP will be worth something. Under these circumstances the rational choice is to hold the IP indefinitely, which is likely what the creditor will do. (In practice, as you pointed out it makes much more sense to just donate it, but given their mindset/incentives, it's extremely unlikely they'd go that route)

**Comment by coffeemug** [comment]
BTW, I feel I should clarify that this is purely my personal opinion on AGPL; if things work out differently and the leadership team goes a different route, I'll of course support them to the degree that I can. I care about RethinkDB a lot (man, ReQL/clustering are beautiful!) and want the open-rethinkdb project to succeed. My comment above might have been a bit adversarial -- sorry about that! I just personally don't find the anti-AGPL arguments compelling, but I appreciate that other people have different views on this.

**Comment by v3ss0n** [comment]
> sorry about that! I just personally don't find the anti-AGPL arguments compelling,
Those who develop projects for tech savvy clients , need to report all opensource licenses we are using. Especially to those tech savvy clients like Telcos and Lawfirms . We have to report every piece of opensource software we used. They have tech consultants reviewing thoses .  
I have one client that totally Refuse AGPL and GPLv3 licenses . There is no way we can get around that. We cannot reason or educate the client on this. Yes we cannot use MongoDB too.
I am sure that is FUD on their part and Anti (A)GPLv3 , trying to educate them will result in lost of the project and client. 

That is why many of us have anti-AGPL arguments. Not because we hate it or FUD , but because we don't usually have a choice to educate clients that disallow AGPL/GPLv3 . I hope you understand that we are not spreading FUD  .

**Comment by uchuugaka** [comment]
I'm with @bcantrill on this. Any kind of GPL is just toxic. You want money into the project to make it go. Even to maintain it. Nobody can keep up a large project out of love alone forever. Companies are often happy to take up MIT/BSD/Apache and similarly licensed projects and contribute because they can use it themselves without their legal departments saying "don't touch this because A/GPL puts all tangentially connected IP at risk."

**Comment by uchuugaka** [comment]
I would also urge looking at it deeply and giving it time to choose it. MySQL is only as big as it is from 15 years of people thinking the license just meant they didn't need to do anything. (including attribution). I'd love to use this in my workplace, but I cannot with AGPL. Legal would cry loudly, especially with no alternative but AGPL.

**Comment by uchuugaka** [comment]
Well, back to the other databases 👍

**Comment by rushmorem** [comment]
While I deeply respect @coffeemug and understand his opinion on this matter, I feel that imposing such views on the project is not a good move. This is something a lot of community members feel strongly about. It has been voiced out time and again even before the company behind RethinkDB shutdown. This is certainly not the last time this issue is going to be brought up if nothing is done. @williamstein [wrote about this](https://sagemath.blogspot.com/2016/10/rethinkdb-must-relicense-now-what-is.html) at length. I think now is the best time to act on this especially since this project is going to be community driven.

I personally do not have a problem with the current license. In fact, my company already uses and loves RethinkDB. I'm working on a [driver for Rust](https://github.com/rust-rethinkdb/reql) as we speak. However, I also understand the points of view raised by both sides on this matter. Let's not try and fight our clients, it's not a fight we can win if we want to survive.

Yes licensing is not the only thing that determines the success of a project but neither is just continued development and velocity if no one is willing to fund the project. It's a combination of various factors. A very good, and practical reason that has been raised for changing the license is the ability to join various foundations.

Some people seem to think that @coffeemug has dismissed the possibility of re-licensing entirely. However, this is not the case. I mean he did say:-

> BTW, I feel I should clarify that this is purely my personal opinion on AGPL; if things work out differently and the leadership team goes a different route, I'll of course support them to the degree that I can.

Also, as he pointed out earlier on:-

> There is a difference between licensing and IP.

While trying to acquire IP rights sounds futile at this time, licensing is a different issue. Why don't we tackle that for now instead. Even without the IP rights that should open some opportunities for us, right?

**Comment by v3ss0n** [comment]
> I think now is the best time to act on this especially since this project is going to be community driven.
I agree.There are core devs still maintaining this project. Nearing next release. I hope people do not sway away . We will find a way around to fund this project, even tho options are quite limited now.
Let discuss on how we can move forward, how we can setup to fund remaining core developers.

**Comment by sirinath** [comment]
I think a petition to the IP holder could be tried though it might fail. Any way how owns the IP now?

**Comment by v3ss0n** [comment]
@sirinath  not sure @coffeemug  can say this but :
here are the list of investors of rethinkdb https://www.crunchbase.com/organization/rethinkdb/investors

**Comment by streaky** [comment]
> but non-zero chance that some day the IP will be worth something

It's not a non-zero chance if the project dies a slow death because nobody can/will contribute - or nobody will use it. The company died for a reason, and that reason is directly related to the viability of the company to make money for its owners. That's obviously a worst-case scenario though - the IP does nothing for anybody in that scenario - what I would say is if somebody can convince the copyright holders that a freer license makes it more likely that some day that IP might be worth something (which could well be true) then maybe they would be willing to relicense which at the end of the day costs them effectively nothing.

For sure though no developer was ever harmed in the making of AGPL software (quite the opposite, it's a more developer-friendly license than the plain old GPL which is precisely why it exists) - and no corporate user was ever harmed in the usage of AGPL software - where it gets slightly complicated is if you're a modifier of software to fit your own needs and _most_ people who do that will have no problem with the provisions if they understand why they exist; it's only very very complicated if you're lifting out _portions_ of code to use inside some other software be it open source or some closed commercial project - but I'm not entirely sure why anybody would do that with rethinkdb's code anyway.

As long as enough people are willing to contribute I don't see what the issue is really though. I'd be wary of getting bogged down as long as this holds true.

**Comment by ha1331** [comment]
@sirinath let's not petition anyone just yet. At least consult the leadership team before you do anything. I mean they are negotiating with these people, worse thing I can imagine anyone can do is to just start spamming random investors and petition for free stuff. Not dictating what people should do, just hoping no one starts an enthusiastic solo operation to get stuff done when there are people who are doing their best to negotiate best possible outcome for all. If you want to participate and share ideas, I suggest you share these plans with say @mglukhovsky @KittyBot @marshall007 and then decide how to proceed.

**Comment by rushmorem** [comment]
I think right now the most pressing issue is that of licensing. @coffeemug thank you for your offer to help:-

> if things work out differently and the leadership team goes a different route, I'll of course support them to the degree that I can.

We highly appreciate that. I'm not part of the leadership but I'm a very interested stakeholder (being a consumer of this excellent database system as well as a community member). What I have observed is that this community is strongly in favour of re-licensing. As such, I would like to appeal to @coffeemug to kindly assist our leadership in re-licensing to `Apache License Version 2.0` or any other license that the community deems fit.

**Comment by mglukhovsky** [comment]
Please check in with the Open RethinkDB leadership team before taking any extraordinary steps. We care a great deal about preserving RethinkDB's open-source future (and its licensing.) The enthusiasm is appreciated, but we can best accomplish an open-source future by coordinating our efforts and working together on this.

(speaking as one of the founders of RethinkDB, the company, and on the leadership team.)

**Comment by rosskukulinski** [comment]
This thread makes me so excited for the future of RethinkDB.  I've been working closely with @mglukhovsky, @kittybot, @marshall007, @atnnn, and others behind the scenes working through all of these details.  We've got the contacts with the right people in the right places and I promise you we are pushing hard to make progress towards a truly open RethinkDB.

**Comment by williamstein** [comment]
> "Worrying about licensing is what PG would call a sitcom idea[1] -- it feels like doing useful work, but in actuality it makes no difference whatsoever."

Reading this thread makes me glad that as of today https://cloud.sagemath.com no longer uses RethinkDB at all.  I spent the last two months doing a rewrite using PostgreSQL.   I hope to have time to write a blog post about what this involved...

**Comment by v3ss0n** [comment]
@williamstein  for the best interest of rethinkdb community efforts , and especially if you respect the community and ex team members trying hard , please do not sway the community like this.
I understand your faustrations , the team is working hard to solve them . So lets not spread FUD .

**Comment by williamstein** [comment]
@v3ss0n -- Sorry, I will have nothing further to do with the RethinkDB community itself.  Goodbye.

**Comment by v3ss0n** [comment]
@williamstein  thank you , then , can you delete your previous post since you no longer interest in this community.

**Comment by bcantrill** [comment]
I don't think that @williamstein should delete his previous comments, because his feelings for this come from the painful spot of needing to retool his own software to avoid a RethinkDB dependency;  those who dismiss the magnitude licensing issue should see and appreciate this pain.  At the same time, I would echo what @v3ss0n, @rosskukulinski and @mglukhovsky have said:  there is a group of folks working very hard to get this resolved, and we need everyone to sit tight.  So @williamstein: we hear you, and hope to have a RethinkDB in short order that you can return to -- and others who might be in his situation, please hang on just a little longer!

**Comment by uchuugaka** [comment]
I think it is deeply appreciated from all. 

I am glad to see how passionate everyone is for this on all sides. Ultimately it shows there is a bit of software everyone sees enough value in to be interested in it. 
I hope RethinkDB carries on one way or the other (my preferred way or not) time is on its side and it will gain more traction by virtue of the quality of features and implementation and care put in.

**Comment by grantcarthew** [comment]
@bcantrill Thanks very much for this input. I have a [project](https://www.npmjs.com/package/rethinkdb-job-queue) that took six months or more to build and it will all be for naught if RethinkDB is abandoned. Please keep us informed.

**Comment by mikeal** [comment]
Having been through the "OMG we have to start a foundation" ringer, I'll give a big +1 to joining CNCF. Solid governance and funding and a lot of projects that are quite complimentary to RethinkDB.

Regarding licensing, the concerns over AGPL are real. Adoption is greatly hindered by that license and the project would be much better off with something more acceptable to more users. If you want to optimize for an ideology about software freedom, that's the correct license, but if you want to optimize for adoption, which you can then feed into a health contributor base, you're better off with something usable by more users.

**Comment by uchuugaka** [comment]
I have to concur. 
At my company any form of GPL and they will happily pay for a commercial license to avoid GPL. It's not about the money. But if it's not easily done they'll walk away.

**Comment by srh** [comment]
If your company won't use Linux, they won't have much use for RethinkDB anyway.

**Comment by mikekidder** [comment]
@srh there are binaries to run in Windows, and Docker containers.  That makes it appealing for us.

**Comment by uchuugaka** [comment]
There are more use cases than for Linux web servers. 
It is also built for macOS. 
It's also BUILDABLE by anybody (technically if not legally). 
Lots of fascinating uses for db that provides these abilities. 
You'd be amazed when it comes to a better logstash or assembly line QC conditions down to the part.

**Comment by streaky** [comment]
> any form of GPL and they will happily pay for a commercial license to avoid GPL

Well, good luck with that. Your company must love not dealing with 90% of the software industry.

No kidding if you're afraid of the GPL why even look at open source projects, much less waste the energy commenting on things.

**Comment by sandstrom** [comment]
Our company is using MongoDB and we considered switching to RethinkDB. Licensing isn't something we spend much time worrying about (GPL/AGPL/MIT differences).

I love RethinkDB, but the main thing that has kept us from switching so far is fears around adoption and lack of a mature ORM in our language (Mongodb has Mongoid for Ruby) — both related to developer contributions and ecosystem liveliness. *Just an anecdote though.*

**Comment by ShaneCurcuru** [comment]
Congrats on getting the best of both worlds: switching to Apache-2.0, and finding a foundation host your project with some funding too!  Great resolution, best of luck.

**Comment by AlekSi** [comment]
https://www.joyent.com/blog/the-liberation-of-rethinkdb

**Comment by dvergeylen** [comment]
TLDR;

>I am pleased to announce that the CNCF has purchased the intellectual property (code, docs, artwork, trademarks, domain names, etc.) of RethinkDB, Inc. And, of course and as one might imagine, we have also relicensed RethinkDB to be under the Apache Public License 2.0.

From https://www.joyent.com/blog/the-liberation-of-rethinkdb link given by @AlekSi

:tada:

**Comment by mglukhovsky** [comment]
Hey everyone, I've got some wonderful news to share: RethinkDB is [joining The Linux Foundation](https://rethinkdb.com/blog/rethinkdb-joins-linux-foundation/)!

The CNCF generously acquired and [donated RethinkDB's assets to the LF](https://www.joyent.com/blog/the-liberation-of-rethinkdb), allowing us to relicense the project to ASLv2.

Now that the legal hurdles have been cleared, many people have asked how they can contribute to RethinkDB's open-source future. You can help out with contributions and donations here: https://rethinkdb.com/contribute/

Stripe has generously offered to match the first $25k in donations (which will help go towards server costs and funding future development.)

We're looking forward to building RethinkDB's next steps together. 🎉

**Comment by uchuugaka** [comment]
HoooorAaaaaaay

**Comment by sirinath** [comment]
Great news!

**Comment by kvz** [comment]
This came out of nowhere! Congrats to everyone involved, i can see a very bright future for rethinkdb once more 💪🎉

**Comment by v3ss0n** [comment]
Awesome Work , Thank you very much for this aweome fight guys! Thank you very much for listening to Our concerns!
